server/src/main/java/io/crate/role/Role.java

```diff
@@ -189,16 +245,48 @@ public Role with(Set<Privilege> privileges) {
     }
 
     public Role with(@Nullable SecureHash password,
-                     @Nullable JwtProperties jwtProperties) {
+                     @Nullable JwtProperties jwtProperties,
+                     boolean resetSessionSettings,
+                     Map<String, Object> sessionSettings) {
+        Map<String, Object> updatedSessionSettings = getUpdatedSessionSettings(resetSessionSettings, sessionSettings);
```

Comment:
It's a bit weird that for password & jwtProperties the `reset` is resolved on the call-site, and for the session settings it's resolved inside.

Regular `with` pattern would be that it behaves more like a plain data constructor - with no additional logic. So this probably shouldn't have a `resetSessionSettings` parameter.

Reply:
Makes sense, yes, I've changed it.